evaluator/src/main/kotlin/osadl/CompatibilityMatrix.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2022 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.evaluator.osadl
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeFromStream
+
+@Serializable
+data class CompatibilityMatrix(
+    val licenses: List<CompatibilityRow>
+)
+
+@Serializable
+data class CompatibilityRow(
+    val name: String,
+    val compatibilities: List<CompatibilityInfo>
+)
+
+@Serializable
+data class CompatibilityInfo(
+    val name: String,
+    val compatibility: Compatibility,
+    val explanation: String
+)
+
+val matrix by lazy {
```

Comment:
I believe it would be nice to introduce a class for a compatibility matrix which internally 
holds a map, which provides one function to access the information of any cell.

That class could the have a companion object constant for the osadl matrix, which has the code for deserializing the `osadl-matrix-with-explanations.json`. So, the de-serialization / data classes would become private. That would minimize the API expose to the rules a bit.

Would that make sense to you as well?

Reply:
I've introduced such a new class. Its lookup function now also handles the case of licenses that are not present in the matrix, incl. empty ones.

Reply:
Thanks!